pkg/controller/registry/registry_client.go

```diff
@@ -97,25 +100,56 @@ func (rc *Client) FindBundleThatProvides(ctx context.Context, group, version, ki
 	return bundle, nil
 }
 
-// FilterChannelEntries filters out a channel entries that provide the requested
+// FilterChannelEntries filters out channel entries that provide the requested
 // API and come from the same package with original operator and returns the
-// first entry on the list
-func (rc *Client) filterChannelEntries(it *ChannelEntryIterator, excludedPkgName string) *opregistry.ChannelEntry {
+// first entry on the list from the default channel of that package
+func (rc *Client) filterChannelEntries(ctx context.Context, it *ChannelEntryIterator, excludedPkgName string) (*opregistry.ChannelEntry, error) {
+	defChannels := make(map[string]string, 0)
+
 	var entries []*opregistry.ChannelEntry
 	for e := it.Next(); e != nil; e = it.Next() {
 		if e.PackageName != excludedPkgName {
-			entry := &opregistry.ChannelEntry{
-				PackageName: e.PackageName,
-				ChannelName: e.ChannelName,
-				BundleName:  e.BundleName,
-				Replaces:    e.Replaces,
+			// keep track of the default channel for each package
+			if _, ok := defChannels[e.PackageName]; !ok {
+				defChannel, err := rc.getDefaultPackageChannel(ctx, e.PackageName)
+				if err != nil {
+					return nil, err
+				}
+				defChannels[e.PackageName] = defChannel
+			}
+
+			// only add entry to the list if the entry is in the default channel
+			if e.ChannelName == defChannels[e.PackageName] {
+				entry := &opregistry.ChannelEntry{
+					PackageName: e.PackageName,
+					ChannelName: e.ChannelName,
+					BundleName:  e.BundleName,
+					Replaces:    e.Replaces,
+				}
+				entries = append(entries, entry)
 			}
-			entries = append(entries, entry)
 		}
 	}
 
 	if entries != nil {
-		return entries[0]
+		return entries[0], nil
+	}
+	return nil, nil
+}
+
+// GetDefaultPackageChannel uses registry client to get the default
+// channel name for a given package name
+func (rc *Client) getDefaultPackageChannel(ctx context.Context, pkgName string) (string, error) {
+	pkg, err := rc.Client.Registry.GetPackage(ctx, &registryapi.GetPackageRequest{Name: pkgName})
+	if err != nil {
+		return "", err
+	}
+
+	var defChan string
+	if pkg != nil {
+		defChan = pkg.DefaultChannelName
+	} else {
+		return "", fmt.Errorf("package %s not found in registry", pkgName)
 	}
-	return nil
+	return defChan, nil
```

Comment:
This separate declaration and assignment looks vestigal. Can you change this to return the error if `pkg == nil`, drop the `else`, and `return pkg.DefaultChannelName, nil` to make the control flow a little clearer?
```